src/include/txn_inline.h

```diff
@@ -972,6 +975,34 @@ __wt_txn_read_upd_list_internal(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt,
             continue;
 
         WT_ORDERED_READ(prepare_state, upd->prepare_state);
+
+        /*
+         * We previously found a prepared update, check if the update has the same transaction id,
+         * if it does it must not be visible as it is part of the same transaction as the previous
+         * prepared update.
+         */
+        if (prepare_txnid != WT_TXN_NONE && upd->txnid == prepare_txnid) {
+            /*
+             * If we see an update with prepare resolved this indicates that the read, which is
+             * configured to ignore prepared updates raced with the commit of the same prepared
+             * transaction. Increment a stat to track this.
+             *
+             * This case exists as reconciliation chooses which update to read to disk in a newest
+             * to oldest fashion, and cannot handle prepared update resolution happening in the same
```

Comment:
Instead of "cannot handle" could we say "and if prepared update resolution happens in the same direction some artifacts of a prepared transaction could be written to disk while some remain only in-memory."

Reply:
[a79ea33](https://github.com/wiredtiger/wiredtiger/pull/9284/commits/a79ea333d75d330a5a84dd5bd20e544b7e325433)